File: services/shared/src/main/java/com/batu/shared/dto/response/TransactionDetailedCategoryDto.java
package com.batu.shared.dto.response;

import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class TransactionDetailedCategoryDto {

    private UUID transactionDetailedCategoryId;
    private String displayName;
    private String detailedCode;
    private TransactionPrimaryCategoryDto primaryCategory;
}

File: transaction-service/src/main/kotlin/com/batu/transaction_service/dto/TransactionDetailedCategoryDto.kt
package com.batu.transaction_service.dto

import java.util.UUID
import com.batu.transaction_service.dto.TransactionPrimaryCategoryDto

data class TransactionDetailedCategoryDto(
    val transactionDetailedCategoryId: UUID?,
    val displayName: String,
    val detailedCode: String,
    val primaryCategory: TransactionPrimaryCategoryDto
)
